src/common/ManagedTelemetry/Telemetry/Events/EventBase.cs

```diff
@@ -12,5 +13,26 @@ namespace Microsoft.PowerToys.Telemetry.Events
     public class EventBase
     {
         public bool UTCReplace_AppSessionGuid => true;
+
+        private string _version;
+        public string Version
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(_version))
+                {
+                    _version = GetVersionFromAssembly();
+                }
+                return _version;
+            }
+        }
+
+        private string GetVersionFromAssembly()
+        {
+            // For consistency this should be formatted the same way as 
+            // https://github.com/microsoft/PowerToys/blob/710f92d99965109fd788d85ebf8b6b9e0ba1524a/src/common/common.cpp#L635
+            var version = Assembly.GetExecutingAssembly().GetName().Version;

```

Comment:
Maybe guard it with try/catch just to be sure, otherwise it looks good!

Reply:
@martin

> This looks OK now, I will make sure if ColorPicker.exe has a proper version set in, but in general this is the way to go!

@martinchrzan .  Good call I think you will need to add the version props file reference to the colorpickerui as per below. 

`<Import Project="..\..\..\installer\Version.props" />`

Reply:
Sorry commented on the wrong post.  I'm not sure what I would catch here, but I can be more defensive with them null checks.   I'll update.

Reply:
Yeah, seems like it should not throw any exception, null checks are enough